### PR TITLE
Feature/rpc api

### PR DESCRIPTION
**Pull request by rekksson**
Contains:
- _Cap'n Proto_ RPC
- _gen_ folder for *.capnp files
- Application state client API (Client part), which allows adaptive applications to report its state
- Machine state client API (Client part), which allows MSM -> EM communication
- Machine state manager which has simple CLI to change machine state
- Execution management which inherits ApplicationStateManagement & MachineStateManagement server sides
- A simple application which sends state change every second
- _add_adaptive_application_ CMake function to build adaptive application right

### CODE REVIEW

File: api/application_state_client/include/application_state_client.h [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-    kInitializing,
-    kRunning,
-    kShuttingdown
+public:
+  ApplicationStateClient();
+  ~ApplicationStateClient() = default;
+
+  using ApplicationState = ::ApplicationStateManagement::ApplicationState;
+
+  void reportApplicationState(ApplicationState state);
```
Comment (adklimkiew) [inline api/application_state_client/include/application_state_client.h]: ReportApplicationState

File: api/application_state_client/include/application_state_client.h [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
- */
-enum class ApplicationState
+class ApplicationStateClient
 {
-    kInitializing,
-    kRunning,
-    kShuttingdown
+public:
+  ApplicationStateClient();
+  ~ApplicationStateClient() = default;
```
Comment (adklimkiew) [inline api/application_state_client/include/application_state_client.h]: not needed

File: api/application_state_client/src/application_state_client.cpp [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#include "application_state_client.h"
+
+namespace api {
+
+ApplicationStateClient::ApplicationStateClient()
+  : client("unix:/tmp/execution_management", 5923)
```
Comment (adklimkiew) [inline api/application_state_client/src/application_state_client.cpp]: magic strings and numbers

File: api/application_state_client/src/application_state_client.cpp [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+{}
+
+void ApplicationStateClient::reportApplicationState(ApplicationState state)
+{
+  auto& waitScope = client.getWaitScope();
+
+  ApplicationStateManagement::Client cap =
```
Comment (adklimkiew) [inline api/application_state_client/src/application_state_client.cpp]: auto will not work?

File: api/execution_management_api/src/execution_management_api.cpp [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#include "execution_management_api.hpp"
+
+int hello()
+{
```
Comment (adklimkiew) [inline api/execution_management_api/src/execution_management_api.cpp]: what is this for?

File: api/machine_state_client/include/machine_state_client.h [hunk 6]
```diff
@@ -1,17 +1,32 @@
...
+
+  // K_SUCCESS
+  // K_INVALID_STATE
+  // K_INVALID_REQUEST
+  // K_TIMEOUT
+  using StateError = MachineStateManagement::StateError;
+
+  StateError registerClient(std::string appName, std::uint32_t timeout);
+  StateError getMachineState(std::uint32_t timeout, std::string& state);
+  StateError setMachineState(std::string state, std::uint32_t timeout);
```
Comment (adklimkiew) [inline api/machine_state_client/include/machine_state_client.h]: method names inconsistent with api

File: api/machine_state_client/src/machine_state_client.cpp [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+namespace api {
+
+namespace {
+  const auto exceptionLambda = [](kj::Exception&& exception)
+    {
+      switch(exception.getType())
+        {
+        case kj::Exception::Type::OVERLOADED:
+        case kj::Exception::Type::DISCONNECTED:
+          return MachineStateClient::StateError::K_TIMEOUT; break;
```
Comment (adklimkiew) [inline api/machine_state_client/src/machine_state_client.cpp]: break not needed

File: api/machine_state_client/src/machine_state_client.cpp [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+          return MachineStateClient::StateError::K_INVALID_REQUEST;
+        }
+    };
+}
+
+MachineStateClient::MachineStateClient(std::string path)
+  : client(path),
+    timer(client.getIoProvider().getTimer()),
+    clientApplication(client.getMain<MachineStateManagement>())
+{}
```
Comment (adklimkiew) [inline api/machine_state_client/src/machine_state_client.cpp]: wrong initialization order

File: api/execution_management_api/src/execution_management_api.cpp (hunk 5 above)
Comment (rekksson) [reply api/execution_management_api/src/execution_management_api.cpp]: Temporary changes. 
_execution_management_api_ folder deleted

**Pull request merged**